### PR TITLE
DCS-2222 add unduly lenient sentence final check

### PR DESCRIPTION
**Pull request by louisenorris**
Check page and warning added to CA review
Screenshot 2023-05-10 at 16 18 54
Screenshot 2023-05-10 at 16 18 43

### CODE REVIEW

File: server/routes/config/finalChecks.js [hunk 1]
```diff
@@ -56,6 +56,21 @@ module.exports = {
         },
       },
     ],
+    nextPath: {
```
Comment (GurnankCheema) [inline server/routes/config/finalChecks.js:59]: nice

File: server/data/licenceTypes.ts [hunk 2]
```diff
@@ -120,6 +120,7 @@ export interface FinalChecks {
   postpone?: { decision?: YesNo; postponeReason?: string }
   refusal?: { decision?: YesNo; outOfTimeReasons?: string; reason?: string }
   seriousOffence?: { decision?: YesNo }
+  undulyLenientSentence?: { decision?: YesNo }
```
Comment (GurnankCheema) [inline server/data/licenceTypes.ts:123]: nice

File: server/services/licence/licenceStatus.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export = function getLicenceStatus(licenceRecord): LicenceStatus {
         seriousOffenceCheck: TaskState.UNSTARTED,
         onRemandCheck: TaskState.UNSTARTED,
         confiscationOrderCheck: TaskState.UNSTARTED,
+        undulyLenientSentenceCheck: TaskState.UNSTARTED,
```
Comment (GurnankCheema) [inline server/services/licence/licenceStatus.ts:36]: ✅

File: licences-specs/src/test/resources/licences/decision/refused.json [hunk 4]
```diff
@@ -90,6 +90,9 @@
       },
       "confiscationOrder": {
         "decision": "No"
+      },
```
Comment (GurnankCheema) [inline licences-specs/src/test/resources/licences/decision/refused.json:93]: ✅

File: server/views/finalChecks/undulyLenientSentence.pug [hunk 5]
```diff
@@ -0,0 +1,15 @@
+extends ../formTemplates/formTemplate
```
Comment (GurnankCheema) [inline server/views/finalChecks/undulyLenientSentence.pug:1]: ✅

File: server/views/review/finalChecks.pug [hunk 6]
```diff
@@ -35,4 +35,7 @@ block content
             span.block Comments
             span#comments.block.bold
               | #{finalChecks.confiscationOrder.comments}
-
+
```
Comment (GurnankCheema) [inline server/views/review/finalChecks.pug:38]: while I don't doubt this is correct, I can't see where this page is used in the journey. Is it an orphaned page that is no longer used?

File: server/statistics/LicenceStatisticsCollector.ts [hunk 7]
```diff
@@ -138,7 +140,12 @@ export class LicenceStatisticsCollector implements RowConsumer<LicenceRow> {
 
   private countFailedFinalChecks(row: LicenceRow) {
     const { licence } = row
-    if (isOnRemand(licence) || hasConfiscationOrder(licence) || hasSeriousOffence(licence)) {
+    if (
+      isOnRemand(licence) ||
+      hasConfiscationOrder(licence) ||
+      hasSeriousOffence(licence) ||
+      hasUndulyLenientSentence(licence)
```
Comment (GurnankCheema) [inline server/statistics/LicenceStatisticsCollector.ts:147]: ✅

File: server/services/licence/licenceStatus.ts [hunk 8]
```diff
@@ -226,7 +231,13 @@ function getCaStageState(licence) {
       seriousOffenceCheck,
       onRemandCheck,
       confiscationOrderCheck,
-      finalChecks: getOverallState([seriousOffenceCheck, onRemandCheck, confiscationOrderCheck]),
+      undulyLenientSentenceCheck,
+      finalChecks: getOverallState([
+        seriousOffenceCheck,
+        onRemandCheck,
+        confiscationOrderCheck,
+        undulyLenientSentenceCheck,
```
Comment (GurnankCheema) [inline server/services/licence/licenceStatus.ts:239]: ✅

**Review by GurnankCheema — APPROVED** [review]
LGTM

File: server/views/review/finalChecks.pug (hunk 6 above)
Comment (louisenorris) [reply server/views/review/finalChecks.pug:38]: It's pulling through final check info to be visible to the DM when they view this section from the tasklist before making their final decision. I've added a screen shot of the page content for this above.

**Pull request merged**